### PR TITLE
Add end to end endpoint for guui rendering on article

### PR DESCRIPTION
**Pull request by MatthewJWalls**
## What does this change?

If you add ?guui to the end of an article url, it will render the given article using the remote node app. This code is for the tech-time on the 27th of April, we'll refactor it into something more generic after that.

I didn't want to muck around with the return types and method signatures of the existing code, so I just wrote some new methods and added a conditional at the top-most renderArticle function, for easy removal.

I think the edge cache and router should pass the ?guui parameter through fine without any changes, since we handle the .json?guui endpoint and the regex for that is just on '?guui'

## Does this affect other platforms - Amp, Apps, etc?

No

### CODE REVIEW

**Comment by MatthewJWalls** [comment]
@nicl @QuarpT

**Comment by SiAdcock** [comment]
Do you think we ought to be capturing performance metrics, the way [we currently do with Moon?](https://github.com/guardian/frontend/blob/master/common/app/common/metrics.scala#L136). If you don't think it's appropriate in this PR, we should certainly add a card to the Trello board!

File: article/app/controllers/ArticleController.scala [hunk 1]
```diff
@@ -166,11 +171,51 @@ class ArticleController(contentApiClient: ContentApiClient, val controllerCompon
     }
   }
 
+
+  def remoteRenderArticle(payload: String): Future[String] = ws.url(Configuration.moon.moonEndpoint)
```
Comment (SiAdcock) [inline article/app/controllers/ArticleController.scala]: Maybe we could choose a better name for this configuration / endpoint

```scala
Configuration.rendering.renderingEndpoint
```

?

**Comment by SiAdcock** [comment]
In the future, we're thinking of using the name `guui` to refer to the JS rendering library and its components, rather than the rendering service and its associated endpoint. Maybe we can rename variables / methods that contain `guuiXyz` to `renderingXyz`?

File: article/app/controllers/ArticleController.scala [hunk 2]
```diff
@@ -166,11 +171,51 @@ class ArticleController(contentApiClient: ContentApiClient, val controllerCompon
...
+    }
+
+  }
+
+  // for the GUUI demo. Same as mapModel except the render method should return a  Future[Result] instead of Result
+  def mapModelGUUI(path: String, range: Option[BlockRange] = None)(render: PageWithStoryPackage => Future[Result])(implicit request: RequestHeader): Future[Result] = {
+
+    lookup(path, range).map((itemResp: ItemResponse) => responseToModelOrResult(range)(itemResp)).recover(convertApiExceptions).flatMap {
+      case Left(model) => render(model)
+      case Right(other) => Future{RenderOtherStatus(other)}
```
Comment (QuarpT) [inline article/app/controllers/ArticleController.scala]: use `Future.successful` for lifting to `Future`

**Review by QuarpT — APPROVED** [review]

**Comment by MatthewJWalls** [comment]
> Do you think we ought to be capturing performance metrics, the way we currently do with Moon? If you don't think it's appropriate in this PR, we should certainly add a card to the Trello board!

I think this might be worth doing, but I'd like to do it under a different PR since this one is almost ready to go

**Review by SiAdcock — APPROVED** [review]

**Pull request merged**